datafusion/optimizer/src/simplify_expressions/regex.rs

```diff
@@ -124,7 +142,8 @@ fn lower_simple(mode: &OperatorMode, left: &Expr, hir: &Hir) -> Option<Expr> {
         HirKind::Empty => {
             return Some(mode.expr(Box::new(left.clone()), "%".to_owned()));
         }
-        HirKind::Literal(Literal::Unicode(c)) if is_safe_for_like(*c) => {
+        HirKind::Literal(l) => {
```

Comment:
the API has changed from having a Unicode char to a `Box<u8>` so I updated this code to match